Structure_Engine/Query/AnalyticalBars.cs

```diff
@@ -101,7 +101,12 @@ private static List<Bar> AnalyticalBars(BHP.FramingProperties.ConstantFramingPro
 
             ISectionProperty section;
 
-            if (convertedProps.ContainsKey(property))
+            if (property == null)
+            {
+                Engine.Base.Compute.RecordError("One or more properties are null. Please make sure all of the inputs have a value.");
+                return new List<Bar>();
+            }
```

Comment:
I would put this as its own if statement after line 94 @albinber - as further in the method has shown (from your discussion with me on teams) that the `property` should not be null at any point really, which makes sense. So checking if it is null at the start of the method will avoid any crashes later one.